lib/server.js

```diff
@@ -1,73 +1,110 @@
-const express = require('express');
-const fs = require('fs');
-const quote = require('regexp-quote');
-const path = require('path');
-const manifest = require('../ssr/ssr-manifest.json');
+require("dotenv").config();
+const express = require("express");
+const fs = require("fs");
+const quote = require("regexp-quote");
+const path = require("path");
+const manifest = require("../ssr/ssr-manifest.json");
 const appPath = path.join(__dirname, "../ssr", manifest["app.js"]);
 const renderer = require(appPath).default;
 const pageSize = 20;
 const versionFile = `${__dirname}/../version.txt`;
-const version = fs.existsSync(versionFile) ? fs.readFileSync(versionFile, 'utf8') : '0';
+const version = fs.existsSync(versionFile)
+  ? fs.readFileSync(versionFile, "utf8")
+  : "0";
+const axios = require("axios");
 
 // Disabled for now because it causes confusion when we update the data
 // const cache = {};
 
-module.exports = async function({ plants, nurseries }) {
+module.exports = async function ({ plants, nurseries }) {
   const port = process.env.PORT || 3000;
   const app = express();
   const publicDir = `${__dirname}/../public`;
   if (!fs.existsSync(publicDir)) {
     fs.mkdirSync(publicDir);
   }
   app.use(express.static(publicDir));
-  app.use('/images', express.static(`${__dirname}/../images`));
-  app.get('/api/v1/plants', async (req, res) => {
+  app.use("/images", express.static(`${__dirname}/../images`));
+  app.use(express.json());
+
+  app.post("/get-vendors", async (req, res) => {
+    const baseUrl = process.env.PAC_API_BASE_URL;
+    const apiKey = process.env.PAC_API_KEY;
+    const url = `${baseUrl}/Plant/FindVendorsForPlantName?plantName=${req.body.plantName}&zipCode=${req.body.zipCode}&radius=${req.body.radius}&limit=5`;
+    let response = await axios.get(url, {
+      headers: { Authorization: "Bearer " + apiKey },
+    });
+    return res.send(response.data);
+  });
+  app.post("/get-city", async (req, res) => {
+    //console.log(req.params)
+    //call vendor/FindZip
+    const baseUrl = process.env.PAC_API_BASE_URL;
+    const apiKey = process.env.PAC_API_KEY;
+    const url = `${baseUrl}/vendor/FindCity?zipCode=${req.body.zipCode}`;
```

Comment:
Same concern re: the need to validate inputs and build query strings with axios.

Reply:
Suggestion taken.  I don't know that validation is necessary however.   A malicious user cannot exploit the API on the other end.   IF the user entered in a non 5 digit zip and didn't get results I'm not sure they would think too poorly of us.  I'll see what we can do on the client side though for that.